src/Components/DateFacet.tsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import {
+  DateFacet as DateFacetType,
+  DateFacetState,
+  buildDateFacet,
+  DateFacetValue
+} from "@coveo/headless";
+import { headlessEngine } from "../Engine";
+import FormLabel from "@material-ui/core/FormLabel";
+import FormControl from "@material-ui/core/FormControl";
+import FormGroup from "@material-ui/core/FormGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import Box from "@material-ui/core/Box";
+
+interface IRangeDateFacetProps {
+  title: string;
+  field: string;
+  delimiter?: string;
+}
+
+export default class DateFacet extends React.Component<
+  IRangeDateFacetProps,
+  {}
+> {
+  private headlessDateFacet: DateFacetType;
+  state: DateFacetState & {
+    inputValue: "";
+  };
```

Comment:
This inputValue appended to the state was to facilitate facetSearch in the other facet. If there's no facet search involved here, you can remove this (and below where you set it in the constructor)